lib/ASTGen/Sources/ASTGen/Macros.swift

```diff
@@ -428,6 +434,12 @@ func expandFreestandingMacro(
     sourceFilePtr: sourceFilePtr, sourceLocationPtr: sourceLocationPtr, type: Syntax.self) else {
     return 1
   }
+  guard let expansion = macroSyntax.asProtocol(
+    FreestandingMacroExpansionSyntax.self
+  ) else {
+    print("not on a macro expansion node: \(macroSyntax.debugDescription)")
```

Comment:
I realize this was already here, but is there something else we could do here instead? We have the diagnostics engine, could we just output a diagnostic? Same with the `NULL source location` above. Also sometimes we return 1 and other times -1, is that on purpose? Not that it's actually used by the caller...

Happy for this to be a separate PR though, or at least an issue so we're tracking it.